uaclient/files.py

```diff
@@ -0,0 +1,344 @@
+import copy
+import json
+import logging
+import os
+from datetime import datetime
+from typing import Dict, Optional
+
+from uaclient import (
+    defaults,
+    event_logger,
+    exceptions,
+    messages,
+    schemas,
+    util,
+)
+from uaclient.machine_token_data import PublicMachineTokenData
+
+event = event_logger.get_event_logger()
+LOG = logging.getLogger(__name__)
+
+
+class UAFile:
+    def __init__(
+        self,
+        name: str,
+        root_directory: str = defaults.DEFAULT_DATA_DIR,
+        private: bool = True,
+    ):
+        self._root_directory = root_directory
+        self._file_name = name
+        self._is_private = private
+        self._path = None
+
+    @property
+    def path(self) -> str:
+        if not self._path:
+            self._path = os.path.join(
+                self._root_directory, self._file_name
+            )  # type: ignore
+        return self._path  # type: ignore
+
+    @property
+    def is_private(self) -> bool:
+        return self._is_private
+
+    def write(self, content):
+        file_mode = (
+            defaults.ROOT_READABLE_MODE
+            if self.is_private
+            else defaults.WORLD_READABLE_MODE
+        )
+        self.create_dir()
```

Comment:
Is create_dir called anywhere else? does it need to be a separate method?

Reply:
I was just making the write method cleaner 😅. So that it is easier to know what is happening in the method.

Reply:
`create_dir` isn't that complex and I think could be inlined here, but not a strong opinion :upside_down_face: